libvirt/domain.go

```diff
@@ -198,11 +198,12 @@ func domainGetIfacesInfo(domain libvirt.Domain, rd *schema.ResourceData) ([]libv
 	return interfaces, nil
 }
 
-func newDiskForCloudInit(virConn *libvirt.Connect, volumeKey string) (libvirtxml.DomainDisk, error) {
+func newDiskForCloudInit(virConn *libvirt.Connect, domainDef *libvirtxml.Domain,
+	volumeKey string) (libvirtxml.DomainDisk, error) {
 	disk := libvirtxml.DomainDisk{
 		Device: "cdrom",
 		Target: &libvirtxml.DomainDiskTarget{
-			Dev: "hda",
+			Dev: fmt.Sprintf("hd%s", diskLetterForIndex(len(domainDef.Devices.Disks))),
```

Comment:
i was thinking here if we can improve error-handling, in case the domain has no disks. but a part of this looks good.